NUnitTests/Lessons/23/LambdaHomework.cs

```diff
@@ -0,0 +1,129 @@
+        }
+
+        [Test]
+        public void Test_FilterEvenNumbers_Function()
+        {
+            var input = new List<int> { 1, 2, 3, 4 };
+            var expected = new List<int> { 2, 4 };
+            var result = FilterEvenNumbers(input);
+            Assert.That(result, Is.EqualTo(expected).AsCollection);
+        }
+
+
+        // TODO: Uncomment and implement lambda functions instead of regular functions
+
+        [Test]
+        public void Test_Add_Function_To_Lambda()
+        {
+            var lambda = (int x, int y) => x + y;
+            var result = lambda(3, 4);
+            Assert.That(result, Is.EqualTo(7), "Add function result is incorrect");
+        }
+
+        [Test]
+        public void Test_Multiply_Function_To_Lambda()
+        {
+            var lambda = (int x, int y) => x * y;
+            var result = lambda(3, 4);
+            Assert.That(result, Is.EqualTo(12), "Multiply function result is incorrect");
+        }
+
+        [Test]
+        public void Test_FilterEvenNumbers_Function_To_Lambda()
+        {
+            Func<List<int>, List<int>> lambda = numbers => numbers.Where(n => n % 2 == 0).ToList();
+            var input = new List<int> { 1, 2, 3, 4 };
+            var expected = new List<int> { 2, 4 };
+            var result = lambda(input);
+            Assert.That(result, Is.EqualTo(expected).AsCollection, "FilterEvenNumbers function is incorrect");
+        }
+
+        [Test]
+        public void Test_Where_Lambda()
+        {
+            var myList = new List<string> { "one", "two", "three", "four" };
+            var filterredList = myList.Where(e => e.Contains('t'));
+            Assert.That(filterredList.Count, Is.EqualTo(2));
+        }
+
+        // TODO: Uncomment and implement without lambda functions
+        
+        [Test]
+        public void Test_Where_NoLambda()
+        {
+            var myList = new List<string> { "one", "two", "three", "four" };
+            var filterredList = new List<string>();
+
+            foreach (var item in myList)
+            {
+                if (item.Contains("t"))
+                {
+                    filterredList.Add(item);
+                }
+            }
+            Assert.That(filterredList.Count, Is.EqualTo(2), $"{filterredList}");
+        }
+        
+        [Test]
+        public void Test_All_Lambda()
+        {
+            var myList = new List<string> { "one", "two", "three", "four" };
+            var notEmpty = myList.All(e => e.Length > 0);
+            Assert.That(notEmpty, Is.True);
+        }
+        // TODO: Uncomment and implement without lambda functions
+
+        [Test]
+        public void Test_All_NoLambda()
+        {
+            var myList = new List<string> { "one", "two", "three", "four" };
+            foreach (var item in myList)
+            {
+                bool result = item.Length > 0;
+                Assert.That(result, Is.True, $"The myList contains empty strings");
+            }
```

Comment:
this is not the equivalent code to the lambda and task has some code which is not supposed to be touched
[Test]
        public void Test_All_NoLambda()
        {
            var myList = new List<string> { "one", "two", "three", "four" };
            bool result

            ...

            Assert.That(result, Is.True);
        }

you need to remove ... and add some code instead, but not to change the place where Assert is (you should not put Assert inside foreach)

Reply:
Test_All_NoLambda fixed

Reply:
great!